Library/Homebrew/gpg.rb

```diff
@@ -1,26 +1,10 @@
-require "utils"
-
 class Gpg
-  def self.find_gpg(executable)
-    which_all(executable).detect do |gpg|
-      gpg_short_version = Utils.popen_read(gpg, "--version")[/\d\.\d/, 0]
-      next unless gpg_short_version
-      gpg_version = Version.create(gpg_short_version.to_s)
-      @version = gpg_version
-      gpg_version >= Version.create("2.0")
-    end
-  end
-
-  def self.executable
-    find_gpg("gpg") || find_gpg("gpg2")
-  end
-
   def self.available?
-    File.executable?(executable.to_s)
+    File.executable?(GPG2Requirement.new.executable.to_s)
   end
 
-  def self.version
-    @version if available?
+  def self.executable
+    GPG2Requirement.new.executable
```

Comment:
This will not work as expected. Fundamentally `Requirement` GPG lookup and install-time lookup need to find GPG differently: the prior needs to search `HOMEBREW_PATH`, the latter `PATH` (based on the `gpg` that was inserted by the requirement). I think this method should just be `which "gpg" || which "gpg2"`.

Reply:
Thanks. To be honest whilst I've kept an eye on Homebrew I'm not as intimately familiar with it as I used to be simply because I don't have the same level of time available as I did once 😅.

This should be okay using `which` because the only usage in core of `Gpg.executable` comes inside `post_install`, but I'll check. The `gpg2` probing is something I dropped from the Requirement side of these changes because I'm no longer all that sure it's necessary.

Reply:
Cool. It would be good to have `Gpg.executable` so it works consistently in `install`, `post_install` and `test` which should be the case with using `which` as-is.

Reply:
> The gpg2 probing is something I dropped from the Requirement side of these changes because I'm no longer all that sure it's necessary.

This suggests to me that we may want to kill this utility method entirely.

Reply:
It seems logical/sensible to me to check we are actually using a 2.x version of GPG still, which can either be enforced via Requirement or simply be always using Homebrew's `gnupg` formula if you don't mind throwing GPGTools support out of the window.